tower/src/limit/concurrency/service.rs

```diff
@@ -26,9 +26,14 @@ pub struct ConcurrencyLimit<T> {
 impl<T> ConcurrencyLimit<T> {
     /// Create a new concurrency limiter.
     pub fn new(inner: T, max: usize) -> Self {
+        Self::new_owned(inner, Arc::new(Semaphore::new(max)))
+    }
+
+    /// Create a new concurrency limiter with a provided owned semaphore
+    pub fn new_owned(inner: T, semaphore: Arc<Semaphore>) -> Self {
```

Comment:
```suggestion
    /// Create a new concurrency limiter with a provided shared semaphore
    pub fn new_shared(inner: T, semaphore: Arc<Semaphore>) -> Self {
```

I guess technically the `Arc` owns the semaphore 😜